midend/interpreter.cpp

```diff
@@ -859,6 +859,16 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         return;
     }
     if (sv->isUnknown()) {
+        if (auto cast = expression->to<IR::Cast>()) {
```

Comment:
This does not handle the case of enums.
Can you please add a test with an enum and see if this needs to change as well?

Reply:
@mbudiu-vmw  We've updated the test to include Enums and casts related to Enums. There is no need to make changes to the code, since in midend there is an `EliminateSerEnums ` pass that replaces Enums with the specified type